src/main/java/meteordevelopment/meteorclient/systems/hud/Hud.java

```diff
@@ -234,6 +246,17 @@ private void onCustomFontChanged(CustomFontChangedEvent event) {
         }
     }
 
+    @EventHandler(priority = EventPriority.LOWEST)
+    private void onOpenScreen(OpenScreenEvent event) {
+        if (hideInMenu.get()) {
+            if (!wasMenuScreen) wasHudHidden = !this.active;
+
+            if (event.screen instanceof GameMenuScreen || event.screen instanceof HandledScreen) this.active = false;
+            else if (!wasHudHidden) this.active = true;
+        }
+        wasMenuScreen = event.screen instanceof GameMenuScreen || event.screen instanceof HandledScreen;
```

Comment:
Why are there 2 fields `wasMenuScreen` and `wasHudHidden` screen? If I am not missing anything then inside this event handler you can check if the new screen is `GameMenuScreen` or `HandledScreen` and if yes then set `this.active` to false and also some other new field to true if the HUD was active. And if the screen is not one of the two then check if the new field is set to true and re-enable the HUD.

Reply:
This was meant for some odd cases when for example player is already in inventory/menu screen and somehow this or toggle hud option was used using code, macros, etc.